nautobot_device_lifecycle_mgmt/metrics.py

```diff
@@ -0,0 +1,140 @@
+"""Nautobot Lifecycle Management plugin application level metrics ."""
+from prometheus_client.core import GaugeMetricFamily
+
+from nautobot_device_lifecycle_mgmt.models import DeviceSoftwareValidationResult, InventoryItemSoftwareValidationResult
+
+
+def metrics_lcm_validation_report_device_type():
+    """Calculate number of devices with valid software by device_type.
+
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    device_validation_model = (
+        DeviceSoftwareValidationResult.objects.values_list("device__device_type__model", flat=True)
+        .distinct()
+        .order_by()
+    )
+
+    valid_device_software_gauge = GaugeMetricFamily(
+        "nautobot_lcm_valid_by_device_type_total",
+        "Number of devices that have valid software by device_type",
+        labels=["device_type"],
+    )
+
+    invalid_device_software_gauge = GaugeMetricFamily(
+        "nautobot_lcm_invalid_by_device_type_total",
+        "Number of devices that have invalid software by device_type",
+        labels=["device_type"],
+    )
```

Comment:
My vote would be to put both of these into one metric and have a valid/invalid label instead.